### PR TITLE
Remove Senlin project code as it retires after 2024.1

### LINKED ISSUE #1267: Remove Senlin role in main after branching `stable/2024.1`
Senlin did not have a release for `stable/2024.1` and got retired.  We already worked on landing the code for it's removal in #1266 but we should remove that in `main` after we branch to not have dead code floating around.

### PR DESCRIPTION
**Pull request by yaguangtang**
fix #1267

### CODE REVIEW

File: charts/memcached/values_overrides/netpol.yaml [hunk 1]
```diff
@@ -44,9 +44,6 @@ network_policy:
         - podSelector:
             matchLabels:
               application: neutron
-        - podSelector:
-            matchLabels:
-              application: senlin
```
Comment (mnaser) [inline charts/memcached/values_overrides/netpol.yaml:49]: Can you please remove undo this change?

**Review by mnaser — CHANGES_REQUESTED** [review]

File: charts/memcached/values_overrides/netpol.yaml (hunk 1 above)
Comment (yaguangtang) [reply charts/memcached/values_overrides/netpol.yaml]: done with a new commit

**Review by mnaser — APPROVED** [review]
